### PR TITLE
Completing PAT-322 auditing of before/after answers for each CIM submission

### PR DESCRIPTION
**Pull request by timharrison-moj**
This completes the A/Cs for PAT-322 by pre-populating the release and review dates and creating an audit row for each time the CIM form is submitted with updated answers. A view of the risk banding forms will not create audit events unless setting the taskCompleted=true for the first time in this section.

### CODE REVIEW

File: server/data/pathfinderClient.js [hunk 1]
```diff
@@ -21,14 +25,42 @@ const createNominal = async (offenderBasic, nominalStatus, referralReason) => {
   return result.rows[0].id
 }
 
-const updateCimReview = async (nominalId, form, username) => {
+const auditSubmittedSection = async (nominalId, form, username, formSectionId) => {
+  const nominal = await getLatestRiskBandingAssessmentByNominalId(nominalId)
+  const prevSectionValue = isNilOrEmpty(nominal.form) ? '' : R.pick([formSectionId], nominal.form)
+  const newSectionValue = isNilOrEmpty(form) ? '' : R.pick([formSectionId], form)
+
+  if (!isNilOrEmpty(prevSectionValue) && !isNilOrEmpty(newSectionValue)) {
```
Comment (garethmdavies) [inline server/data/pathfinderClient.js]: Do we not audit when the section is first saved?

Comment (timharrison-moj) [reply server/data/pathfinderClient.js]: That's an interesting one - as the initial details are pre-populated now, nothing audits that first creation. But when a user edits the section again and if they change anything it creates an audit row with the before and after view. What I noticed is that even viewing for the first time - if the user saves the section without changing anything there is in fact a 'taskCompleted=true' value added and this also trigger the auditing - but only on the first viewing of each section.

File: migrations/20191114130000_alter_audit_add_previous_value.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+exports.up = knex =>
+  Promise.all([
+    knex.schema.alterTable('audit_event', table => {
+      table.dropColumn('event_text')
+    }),
+    knex.schema.alterTable('audit_event', table => {
+      table.text('event_text').nullable()
```
[the remark this answers is not in the document]
Comment (timharrison-moj) [reply migrations/20191114130000_alter_audit_add_previous_value.js]: event_text should never be - I will alter that. The other columns could be - where there is no previous values.

File: server/data/pathfinderClient.js (hunk 1 above)
Comment (timharrison-moj) [reply server/data/pathfinderClient.js]: Sorry - crossed lines on this - you meant the specific bit of code which requires both prev and current values to be populated?  Good point - I'm checking this too now.

**Pull request merged**